src/translation-assistant.js

```diff
@@ -138,16 +139,23 @@ function getMapping(
     if (findRegex === MATH_REGEX) {
         inputs = inputs
             .map((input) => translateMath(input, lang))
+            // NOTE(danielhollas):The assumption here is that
+            // different math notations will not mingle in a single string,
+            // even though in principle a single string can very well contain
+            // both an interval (0,10) and cartesian point (0, 0) for example
+            // Presumably, such cases are rare.
+            .map((input) => maybeTranslateMath(input, outputs.join(' '), lang))
```

Comment:
in this case, `outputs` hold translated math bits from a translator and we're trying to match the notation that they used.

Reply:
Yeah, unfortunately, this is a chicken-or-egg problem, we're in the `getMapping()` function that creates this mapping: translators can change the order of math bits. I guess we could be more creative, but this is the simplest solution. Also, we do not want to support mixed notation within a single string anyways so I don't think this is an issue.

Reply:
Thank you for the explanation Daniel. Could you please add in a comment as well, e.g.:

```
.map((input) =>
    // We pass in the entire template...
    maybeTranslateMath(input, outputs.join(' '), lang))
```

Better yet maybe extract `outputs.join(' ')` above, e.g.:
```
// We provide the entire template...
const translatedTemplate = outputs.join(' '); // Not sure about the name?

inputs = inputs
    .map((input) => translateMath(input, lang))
    .map((input) => maybeTranslateMath(input, translatedTemplate, lang))
```

?

Reply:
Okay, thanks! I've called it `allTranslatedMaths` and did the same in all three instances where we call `maybeTranslateMath`. I've also switched the order: `maybeTranslateMath` should be called before the numbers are localized in `translateMath`. It doesn't matter here, but will be important for coordinates and intervals.